### DISCUSSION TITLE
authorizing onedrive on cloud vm

### DISCUSSION THREAD

**Discussion by Kaushal-Kishore-Gupta** [abraunegg/onedrive#1818] [Q&A] [answered]
I want to use this client to upload files from my cloud linux machine.
However it has only terminal and so I cant open auth URL to get a response one.

how to authorize onedrive in such case

**Comment by abraunegg** [comment, marked as answer]
@Kaushal-Kishore-Gupta 
> However it has only terminal and so I cant open auth URL to get a response one

You do not have to open the auth URL on the same machine where the client is being run from.

**Reply by Kaushal-Kishore-Gupta** [reply]
but then i am getting authrization error.. and when I am doing it in my machine I am easily able to authorize it

**Reply by abraunegg** [reply]
> but then i am getting authrization error

What error?

What client version are you running - `onedrive --version` ?

**Reply by Kaushal-Kishore-Gupta** [reply]
onedrive v2.3.13-1build2

**Reply by Kaushal-Kishore-Gupta** [reply]
I am running same version in both machine (my laptop and cloud vm) however seems to work fine in my laptop but not in vm

**Reply by abraunegg** [reply]
@Kaushal-Kishore-Gupta 
> onedrive v2.3.13-1build2

That is your problem. Please read the installation instructions correctly.

**Reply by abraunegg** [reply]
@Kaushal-Kishore-Gupta 
> I am running same version in both machine (my laptop and cloud vm) however seems to work fine in my laptop but not in vm

You are running an unsupported and obsolete client version on both your cloud system and your local system.

Please read the instructions on how to install this client correctly.

**Reply by Kaushal-Kishore-Gupta** [reply]
Oh ohkay I am sorry for the ignorance in instruction..
